straxen/misc.py

```diff
@@ -276,3 +277,15 @@ def convert_array_to_df(array: np.ndarray) -> pd.DataFrame:
     """
     keys = [key for key in array.dtype.names if array[key].ndim == 1]
     return pd.DataFrame(array[keys])
+
+@export
+def filter_kwargs(func, kwargs):
+    """Filter out keyword arguments that
+        are not in the call signature of func
+        and return filtered kwargs dictionary
+    """
+    params = inspect.signature(func).parameters
+    if any([str(p).startswith('**') for p in params.values()]):
+        # if func accepts wildcard kwargs, return all
+        return kwargs
```

Comment:
Also here is not this, this typical it is easier to ask for forgiveness than permission example:

```
try: 
    f(**keargs):
except:
     # extract kwargs
```
I am fine with both ways.

Reply:
In general i agree but there are two reasons not to use that pattern here:

1. I prefer not to make any assumptions as to the existence of side effects (the function may be writing to disk/changing some shared object/releasing the Kraken etc.)
2. I prefer not to make assumptions as to the run-time of the function, it may be a very expensive operation/spin indefinitely when it gets the wrong kwargs.

Reply:
I do not fully follow. Why would this be important? You are not really calling the function are you? You directly raise: 

```
def test():
    print('Do nothing yo')

test(bla='a')
```
returns:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/tmp/jobs/15223754/ipykernel_112/2655484829.py in <module>
----> 1 test(bla='a')

TypeError: test() got an unexpected keyword argument 'bla'
```
But as I said for me it is fine either way.

Reply:
The problem isnt when the try block fails, its when it succeeds ;) i.e. in cases where the function does accept all the kwargs passed to it.

Reply:
Ah okay I see.